File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {
    public int solution(int n) {
        int answer = 0;
        int start = 1, end = 1, sum = 1;
        
        while (start <= n) {
            if (sum < n) {
                end++;
                sum += end;
            } else if (sum > n) {
                sum -= start;
                start++;
            } else {
                answer++;
                sum -= start;
                start++;
            }
        }
        
        return answer;
    }
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.js
function solution(n) {
  let answer = 0;
  const doubleN = 2 * n;

  for (let k = 1; k * k <= doubleN; k++) {
    if (doubleN % k === 0) {
      let pair = doubleN / k;
     if ((pair - k + 1) % 2 === 0) answer++;
    }
  }

  return answer;
}